src/semantics/expression_builder.ts

```diff
@@ -1084,6 +1084,7 @@ export function newCastValue(
     let value_type = value.effectType;
     if (value_type.kind == ValueTypeKind.ENUM) {
         value_type = (value_type as EnumType).memberType;
+        value.type = value_type;
```

Comment:
Why set `value.type` as the `memberType` here? I think that if we create `CastValue` in l1521-l1528, the value type can be enum type here.

Reply:
I regard the **_Enum_** type as an alias of its **_memberType_** type. Such treatment does seem to be questionable.